<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>PlannerVisualization</class>
 <widget class="QMainWindow" name="PlannerVisualization">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1430</width>
    <height>820</height>
   </rect>
  </property>
  <property name="acceptDrops">
   <bool>true</bool>
  </property>
  <property name="windowTitle">
   <string>Parking Trajectory Visualization Ultrasonic only</string>
  </property>
  <property name="windowIcon">
   <iconset resource="../resources.qrc">
    <normaloff>:/icons/IconPlan.png</normaloff>:/icons/IconPlan.png</iconset>
  </property>
  <property name="animated">
   <bool>true</bool>
  </property>
  <widget class="QWidget" name="centralWidget">
   <property name="sizePolicy">
    <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
     <horstretch>0</horstretch>
     <verstretch>0</verstretch>
    </sizepolicy>
   </property>
   <layout class="QHBoxLayout" name="horizontalLayout">
    <item>
     <widget class="QSplitter" name="splitter">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Expanding" vsizetype="Preferred">
        <horstretch>50</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <property name="baseSize">
       <size>
        <width>0</width>
        <height>0</height>
       </size>
      </property>
      <property name="orientation">
       <enum>Qt::Horizontal</enum>
      </property>
      <property name="childrenCollapsible">
       <bool>false</bool>
      </property>
      <widget class="QGroupBox" name="groupBox">
       <property name="sizePolicy">
        <sizepolicy hsizetype="Preferred" vsizetype="Expanding">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <property name="minimumSize">
        <size>
         <width>100</width>
         <height>0</height>
        </size>
       </property>
       <property name="title">
        <string>Loaded Scenes</string>
       </property>
       <layout class="QVBoxLayout" name="verticalLayout">
        <property name="leftMargin">
         <number>0</number>
        </property>
        <property name="topMargin">
         <number>0</number>
        </property>
        <property name="rightMargin">
         <number>0</number>
        </property>
        <property name="bottomMargin">
         <number>0</number>
        </property>
        <item>
         <widget class="QListView" name="loadedScenesList">
          <property name="enabled">
           <bool>true</bool>
          </property>
          <property name="sizePolicy">
           <sizepolicy hsizetype="Minimum" vsizetype="Expanding">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="minimumSize">
           <size>
            <width>50</width>
            <height>0</height>
           </size>
          </property>
          <property name="maximumSize">
           <size>
            <width>16777215</width>
            <height>16777215</height>
           </size>
          </property>
          <property name="styleSheet">
           <string notr="true">selection-background-color: rgb(20, 170, 255);
  selection-color: rgb(255, 255, 255);</string>
          </property>
          <property name="editTriggers">
           <set>QAbstractItemView::NoEditTriggers</set>
          </property>
          <property name="selectionBehavior">
           <enum>QAbstractItemView::SelectRows</enum>
          </property>
          <property name="movement">
           <enum>QListView::Static</enum>
          </property>
          <property name="uniformItemSizes">
           <bool>false</bool>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="addEMButton">
          <property name="text">
           <string>Add EM</string>
          </property>
          <property name="icon">
           <iconset resource="../resources.qrc">
            <normaloff>:/icons/IconAdd.png</normaloff>:/icons/IconAdd.png</iconset>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="removeEMButton">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="text">
           <string>Remove EM</string>
          </property>
          <property name="icon">
           <iconset resource="../resources.qrc">
            <normaloff>:/icons/IconRemove.png</normaloff>:/icons/IconRemove.png</iconset>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="layoutWidget">
       <layout class="QVBoxLayout" name="verticalLayout_3">
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_2">
          <item>
           <widget class="QLabel" name="pathStepIndex">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Minimum" vsizetype="Minimum">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="text">
             <string>StepIndex</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QSlider" name="pathStepSlider">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Expanding" vsizetype="Minimum">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="tracking">
             <bool>true</bool>
            </property>
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="btnApplyStartPose">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Minimum" vsizetype="Minimum">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="text">
             <string>Set as start pose</string>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <widget class="ParkingSceneWidget" name="parkingWidget" native="true">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
         </widget>
        </item>
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_3"/>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="layoutWidget2">
       <layout class="QVBoxLayout" name="verticalLayout_2" stretch="0">
        <item>
         <widget class="QSplitter" name="splitter_2">
          <property name="orientation">
           <enum>Qt::Vertical</enum>
          </property>
          <widget class="ObjectTreeWidget" name="objectTreeWidget" native="true">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="autoFillBackground">
            <bool>false</bool>
           </property>
          </widget>
          <widget class="QTreeView" name="objectPropertiesView">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
          </widget>
         </widget>
        </item>
       </layout>
      </widget>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1430</width>
     <height>26</height>
    </rect>
   </property>
   <widget class="QMenu" name="menu_File">
    <property name="title">
     <string>&amp;File</string>
    </property>
    <addaction name="actionLoad_Trajectory_Planner_DLL"/>
    <addaction name="actionLoad_Environment_Model"/>
    <addaction name="separator"/>
    <addaction name="actionReload_Last_Scene"/>
    <addaction name="separator"/>
    <addaction name="actionExport_FakeEM_for_car"/>
    <addaction name="actionSave_Json_Reachability"/>
    <addaction name="actionLoad_Json_Reachability"/>
   </widget>
   <widget class="QMenu" name="menuSettings">
    <property name="title">
     <string>Settings</string>
    </property>
    <addaction name="actionLoad_Trajectory_Planner_Parameters"/>
    <addaction name="actionLoad_Vehicle_Parameters"/>
    <addaction name="actionLoad_Sys_Func_Parameters"/>
    <addaction name="separator"/>
    <addaction name="actionSettings"/>
   </widget>
   <widget class="QMenu" name="menuView">
    <property name="title">
     <string>View</string>
    </property>
    <addaction name="actionEnable_Debug_Draw"/>
    <addaction name="actionShow_All"/>
    <addaction name="actionShow_Trajectory_Curvature"/>
    <addaction name="actionDraw_Target_Pose_Reachable_Area_3D"/>
   </widget>
   <widget class="QMenu" name="menuAbout">
    <property name="title">
     <string>Help</string>
    </property>
    <addaction name="actionAbout"/>
   </widget>
   <widget class="QMenu" name="menuTools">
    <property name="title">
     <string>Tools</string>
    </property>
    <addaction name="actionGrab_to_clipboard"/>
   </widget>
   <addaction name="menu_File"/>
   <addaction name="menuView"/>
   <addaction name="menuTools"/>
   <addaction name="menuSettings"/>
   <addaction name="menuAbout"/>
  </widget>
  <widget class="QToolBar" name="mainToolBar">
   <property name="iconSize">
    <size>
     <width>48</width>
     <height>32</height>
    </size>
   </property>
   <property name="toolButtonStyle">
    <enum>Qt::ToolButtonTextUnderIcon</enum>
   </property>
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
   <addaction name="actionLoad_Environment_Model"/>
   <addaction name="separator"/>
   <addaction name="actionStartScanMode"/>
   <addaction name="actionPlan_Path"/>
   <addaction name="actionReplan_Path"/>
   <addaction name="actionMake_Step"/>
   <addaction name="actionJump"/>
   <addaction name="separator"/>
   <addaction name="actionContinuous_Replanning"/>
   <addaction name="actionPlan_GaragePark"/>
   <addaction name="separator"/>
   <addaction name="actionStart_TAPOSD"/>
   <addaction name="separator"/>
   <addaction name="actionUndo"/>
   <addaction name="actionRedo"/>
   <addaction name="separator"/>
   <addaction name="actionEnable_Debug_Draw"/>
   <addaction name="separator"/>
   <addaction name="actionDefine_Reachability_Area"/>
   <addaction name="actionDraw_Target_Pose_Reachable_Area_2D"/>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
  <action name="actionLoad_Trajectory_Planner_Parameters">
   <property name="text">
    <string>Load Trajectory Planner Parameters...</string>
   </property>
  </action>
  <action name="actionLoad_Vehicle_Parameters">
   <property name="text">
    <string>Load Vehicle Parameters...</string>
   </property>
  </action>
  <action name="actionLoad_Sys_Func_Parameters">
   <property name="text">
    <string>Load Sys Func Parameters...</string>
   </property>
  </action>
  <action name="actionLoad_Environment_Model">
   <property name="icon">
    <iconset resource="../resources.qrc">
     <normaloff>:/icons/IconLoad.png</normaloff>:/icons/IconLoad.png</iconset>
   </property>
   <property name="text">
    <string>Load EM...</string>
   </property>
   <property name="toolTip">
    <string>Load JSON Environment Model (Ctrl+O)</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+O</string>
   </property>
  </action>
  <action name="actionLoad_Trajectory_Planner_DLL">
   <property name="text">
    <string>Load Trajectory Planner DLL</string>
   </property>
  </action>
  <action name="actionReload_Last_Scene">
   <property name="icon">
    <iconset resource="../resources.qrc">
     <normaloff>:/icons/IconReload.png</normaloff>:/icons/IconReload.png</iconset>
   </property>
   <property name="text">
    <string>Reload Last EM</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+R</string>
   </property>
  </action>
  <action name="actionPlan_Path">
   <property name="icon">
    <iconset resource="../resources.qrc">
     <normaloff>:/icons/IconPlan.png</normaloff>:/icons/IconPlan.png</iconset>
   </property>
   <property name="text">
    <string>Plan</string>
   </property>
   <property name="toolTip">
    <string>Plan new path (P)</string>
   </property>
   <property name="shortcut">
    <string>P</string>
   </property>
  </action>
  <action name="actionReplan_Path">
   <property name="icon">
    <iconset resource="../resources.qrc">
     <normaloff>:/icons/IconReplan.png</normaloff>:/icons/IconReplan.png</iconset>
   </property>
   <property name="text">
    <string>Replan</string>
   </property>
   <property name="toolTip">
    <string>Replan path (R)</string>
   </property>
   <property name="shortcut">
    <string>R</string>
   </property>
  </action>
  <action name="actionMake_Step">
   <property name="icon">
    <iconset resource="../resources.qrc">
     <normaloff>:/icons/IconStep.png</normaloff>:/icons/IconStep.png</iconset>
   </property>
   <property name="text">
    <string>Step</string>
   </property>
   <property name="toolTip">
    <string>Make step on planned path (S)</string>
   </property>
   <property name="shortcut">
    <string>S</string>
   </property>
  </action>
  <action name="actionUndo">
   <property name="icon">
    <iconset resource="../resources.qrc">
     <normaloff>:/icons/IconUndo.png</normaloff>:/icons/IconUndo.png</iconset>
   </property>
   <property name="text">
    <string>Undo</string>
   </property>
   <property name="toolTip">
    <string>Undo (Ctrl+Z)</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Z</string>
   </property>
  </action>
  <action name="actionRedo">
   <property name="icon">
    <iconset resource="../resources.qrc">
     <normaloff>:/icons/IconRedo.png</normaloff>:/icons/IconRedo.png</iconset>
   </property>
   <property name="text">
    <string>Redo</string>
   </property>
   <property name="toolTip">
    <string>Redo (Ctrl+Y)</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Y</string>
   </property>
  </action>
  <action name="actionSettings">
   <property name="icon">
    <iconset resource="../resources.qrc">
     <normaloff>:/icons/IconSettings.png</normaloff>:/icons/IconSettings.png</iconset>
   </property>
   <property name="text">
    <string>Settings</string>
   </property>
  </action>
  <action name="actionEnable_Debug_Draw">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="icon">
    <iconset resource="../resources.qrc">
     <normaloff>:/icons/IconDebug.png</normaloff>:/icons/IconDebug.png</iconset>
   </property>
   <property name="text">
    <string>Debug Draw</string>
   </property>
   <property name="toolTip">
    <string>Enable Debug Draw (Ctrl+D)</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+D</string>
   </property>
  </action>
  <action name="actionShow_All">
   <property name="text">
    <string>Show All</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+A</string>
   </property>
  </action>
  <action name="actionAbout">
   <property name="text">
    <string>About ...</string>
   </property>
  </action>
  <action name="actionGrab_to_clipboard">
   <property name="text">
    <string>Grab to clipboard</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+G</string>
   </property>
  </action>
  <action name="actionPlan_Geometrical_Path">
   <property name="icon">
    <iconset resource="../resources.qrc">
     <normaloff>:/icons/IconGeometric.png</normaloff>:/icons/IconGeometric.png</iconset>
   </property>
   <property name="text">
    <string>Plan Geom</string>
   </property>
   <property name="toolTip">
    <string>Plan Geometrical Path (Ctrl+P)</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+P</string>
   </property>
  </action>
  <action name="actionStartScanMode">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="icon">
    <iconset resource="../resources.qrc">
     <normaloff>:/icons/IconScanning.png</normaloff>:/icons/IconScanning.png</iconset>
   </property>
   <property name="text">
    <string>Scan</string>
   </property>
   <property name="toolTip">
    <string>Start Scan Mode (Ctrl+S)</string>
   </property>
   <property name="shortcut">
    <string>Shift+S</string>
   </property>
  </action>
  <action name="actionShow_Trajectory_Curvature">
   <property name="text">
    <string>Show Trajectory Curvature and Velocity</string>
   </property>
  </action>
  <action name="actionDraw_Target_Pose_Reachable_Area_2D">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="icon">
    <iconset resource="../resources.qrc">
     <normaloff>:/icons/IconDraw2D.png</normaloff>:/icons/IconDraw2D.png</iconset>
   </property>
   <property name="text">
    <string>Draw Reachability</string>
   </property>
   <property name="toolTip">
    <string>Draw Target Pose Reachable Area in 2D</string>
   </property>
  </action>
  <action name="actionDraw_Target_Pose_Reachable_Area_3D">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="icon">
    <iconset resource="../resources.qrc">
     <normaloff>:/icons/IconDraw3D.png</normaloff>:/icons/IconDraw3D.png</iconset>
   </property>
   <property name="text">
    <string>Reachability Area in 3D</string>
   </property>
   <property name="toolTip">
    <string>Draw Target Pose Reachable Area in 3D</string>
   </property>
  </action>
  <action name="actionStart_TAPOSD">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="icon">
    <iconset resource="../resources.qrc">
     <normaloff>:/icons/IconTaposd.png</normaloff>:/icons/IconTaposd.png</iconset>
   </property>
   <property name="text">
    <string>Target Pose</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+T</string>
   </property>
  </action>
  <action name="actionDefine_Reachability_Area">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="icon">
    <iconset resource="../resources.qrc">
     <normaloff>:/icons/IconDefine.png</normaloff>:/icons/IconDefine.png</iconset>
   </property>
   <property name="text">
    <string>Define Reachabity</string>
   </property>
   <property name="shortcut">
    <string>D</string>
   </property>
  </action>
  <action name="actionExport_FakeEM_for_car">
   <property name="text">
    <string>Export FakeEM for car...</string>
   </property>
  </action>
  <action name="actionContinuous_Replanning">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="icon">
    <iconset resource="../resources.qrc">
     <normaloff>:/icons/IconConRe.png</normaloff>:/icons/IconConRe.png</iconset>
   </property>
   <property name="text">
    <string>Continous Replanning</string>
   </property>
  </action>
  <action name="actionLoad_Json_Reachability">
   <property name="text">
    <string>Load Json Reachability Area...</string>
   </property>
  </action>
  <action name="actionLoad_Measurement">
   <property name="icon">
    <iconset>
     <normaloff>:/icons/IconMTS.png</normaloff>:/icons/IconMTS.png</iconset>
   </property>
   <property name="text">
    <string>Load Measurement...</string>
   </property>
   <property name="toolTip">
    <string>Load MTS .rrec Measurement</string>
   </property>
  </action>
  <action name="actionPlan_GaragePark">
   <property name="icon">
    <iconset resource="../resources.qrc">
     <normaloff>:/icons/IconGaragePark.png</normaloff>:/icons/IconGaragePark.png</iconset>
   </property>
   <property name="text">
    <string>Garage Park</string>
   </property>
   <property name="toolTip">
    <string>Garage Park</string>
   </property>
  </action>
  <action name="actionJump">
   <property name="icon">
    <iconset resource="../resources.qrc">
     <normaloff>:/icons/IconJump.png</normaloff>:/icons/IconJump.png</iconset>
   </property>
   <property name="text">
    <string>Jump</string>
   </property>
   <property name="toolTip">
    <string>Jump to distanceToStop (J)</string>
   </property>
   <property name="shortcut">
    <string>J</string>
   </property>
  </action>
  <action name="actionSave_Json_Reachability">
   <property name="text">
    <string>Save Json Reachability Area...</string>
   </property>
  </action>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <customwidgets>
  <customwidget>
   <class>ParkingSceneWidget</class>
   <extends>QWidget</extends>
   <header>ui/parkingscenewidget.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>ObjectTreeWidget</class>
   <extends>QWidget</extends>
   <header>ui/objecttreewidget.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources>
  <include location="../resources.qrc"/>
 </resources>
 <connections/>
 <slots>
  <slot>onLoadDLL()</slot>
  <slot>onCallInit()</slot>
  <slot>onCallRun()</slot>
  <slot>onPathIndexChanged(int)</slot>
 </slots>
</ui>
